packages/nextjs/components/Search.stories.tsx

```diff
@@ -0,0 +1,126 @@
+import type { Meta, StoryObj } from ".storybook/types";
+import { expect } from "@storybook/jest";
+import { userEvent, waitFor, within } from "@storybook/testing-library";
+
+import { Search } from "./Search";
+
+const meta: Meta<typeof Search> = {
+  component: Search,
+  args: {},
+};
+
+export default meta;
+
+type Story = StoryObj<typeof Search>;
+
+export const Default: Story = {
+  async play({ canvasElement, step }) {
+    const ui = wrap(canvasElement);
+    await step("expect to see a search box", async () => {
+      expect(ui.searchbox).toBeVisible();
+    });
+  },
+};
+
+export const WithSearchTerm: Story = {
+  async play({ canvasElement, step }) {
+    const ui = wrap(canvasElement);
+
+    await step("Type 'baguette' into the search box", async () => {
+      ui.searchbox.focus();
+      await userEvent.type(ui.searchbox, "baguette");
+    });
+
+    await step("expect to see a loading indicator", async () => {
+      expect(ui.resultsBox).toBeVisible();
+      expect(ui.resultsText).toEqual(["Loading..."]);
+    });
+
+    await step("expect to see some search results", async () => {
+      await waitFor(
+        () => {
+          expect(ui.resultsText).toHaveLength(2);
+        },
+        { timeout: 3000 }
```

Comment:
if we're using msw for tests, 3 seconds seems very large?

Reply:
For these Stories we're actually not using MSW, it's making a real Sanity request.  
I would like to follow-up with a blog about MSW, or perhaps piggy-back off Kevin's MSW post.  But for now I just left this connected to the production server, YOLO.